### PR TITLE
User system using the api

### LINKED ISSUE #6: Create a user system in the backend
the system MUST support the following feature : 
 - login a user
 - register  a new user

### PR DESCRIPTION
**Pull request by jorisSchaller**
This PR add the code in the api module to do the correcte action using the db module wrapper.
Here we added the tests and the code related to the creation of a new user and the login of a user using the api.

closes #6

### CODE REVIEW

File: backend/api/api_test.go [hunk 1]
```diff
@@ -1,23 +1,287 @@
...
+	w := httptest.NewRecorder()
+	req := newUserRegisterRequest("username", "passwd")
+	router.ServeHTTP(w, req)
+	assert.Equal(t, http.StatusCreated, w.Code)
+}
+func TestIllFormedLoginFail(t *testing.T) {
+	js := fmt.Sprintf(`
+	{"pwd": "%s"
+	"usr" : "%s" `, "pwd", "usr")
+	req, _ := http.NewRequest("POST", "/user/login", strings.NewReader(js))
```
Comment (Luhrel) [inline backend/api/api_test.go:128]: Ditto

File: backend/api/api_test.go [hunk 2]
```diff
@@ -1,23 +1,287 @@
...
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, req)
+	log.Printf("%s", w.Body)
+	assert.Equal(t, http.StatusBadRequest, w.Code)
+}
+
+func TestIllFormed2LoginFail(t *testing.T) {
+	js := fmt.Sprintf(`
+	{"password": "%s"}`, "pwd")
+	req, _ := http.NewRequest("POST", "/user/login", strings.NewReader(js))
```
Comment (Luhrel) [inline backend/api/api_test.go:138]: Ditto

File: backend/api/api_test.go [hunk 3]
```diff
@@ -1,23 +1,287 @@
...
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, req)
+	log.Printf("%s", w.Body)
+	assert.Equal(t, http.StatusUnauthorized, w.Code)
+}
+
+func TestLoginSucess(t *testing.T) {
+
+	//delete all users
+	client, _ := db.GetDBConnection()
```
Comment (Luhrel) [inline backend/api/api_test.go:148]: Ditto

File: backend/api/api_test.go [hunk 4]
```diff
@@ -1,23 +1,287 @@
...
+	_ = deleteCollection(context.Background(), client, client.Collection("users"), 64)
```
Comment (Luhrel) [inline backend/api/api_test.go:149]: Is `_ =` useless ?

File: backend/db/db.go [hunk 5]
```diff
@@ -28,16 +30,16 @@ type User struct {
 }
 
 // Ther is two type : low and high
-type Profile int
+type Profile uint16
```
Comment (Luhrel) [inline backend/db/db.go:33]: uint8 ?

File: backend/db/db.go [hunk 6]
```diff
@@ -28,16 +30,16 @@ type User struct {
...
 const (
 	low Profile = iota
 	high
+	count
 )
 
 type Sock struct {
-	SockId   string `firestore:"sockId,omitempty" json:"sockId"`
-	ShoeSize int    `firestore:"shoeSize" json:"shoeSize"`
+	ShoeSize uint16 `firestore:"shoeSize" json:"shoeSize"`
```
Comment (Luhrel) [inline backend/db/db.go:42]: `uint8`?

File: backend/db/db_test.go [hunk 7]
```diff
@@ -1,15 +1,64 @@
...
+	"google.golang.org/api/iterator"
 )
 
 func TestMain(m *testing.M) {
 	os.Setenv(FirestoreEmulatorHost, "localhost:8080")
 	log.SetFlags(log.Flags() | log.Llongfile)
-	os.Exit(m.Run())
+	exit := m.Run()
+	// delete collection after run
+	client, _ := GetDBConnection()
```
Comment (Luhrel) [inline backend/db/db_test.go:18]: Should fail on error

**Comment by jorisSchaller** [comment]
I changed as requested 
@Luhrel

**Review by Luhrel — APPROVED** [review]

**Pull request merged**